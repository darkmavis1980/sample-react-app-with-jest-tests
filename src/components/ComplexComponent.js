import React, {Component} from 'react';
import { Button } from 'semantic-ui-react'
import { MakeUppercase, ConsoleStuff } from './Utilities';

class ComplexComponent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      test: 'hello world',
      counter: 0
    }

    this.clickHandler = this.clickHandler.bind(this);
    this.increase = this.increase.bind(this);
    this.decrease = this.decrease.bind(this);
  }

  clickHandler() {
    ConsoleStuff(this.state.test);
  }

  decrease() {
    let counter = this.state.counter;
    counter--;
    this.setState({counter});
  }

  increase() {
    let counter = this.state.counter;
    counter++;
    this.setState({counter});
  }

  render() {
    return (
      <div>
        <h3>{MakeUppercase(this.props.title)}</h3>
        <Button onClick={this.clickHandler}>Click me!</Button>
        <Button onClick={this.decrease}>-</Button>
        <Button onClick={this.increase}>+</Button>
        <p>{this.state.counter}</p>
      </div>
    );
  }
}

export default ComplexComponent;