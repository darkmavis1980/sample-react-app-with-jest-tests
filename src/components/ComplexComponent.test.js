import React from 'react';
import ComplexComponent from './ComplexComponent';

const title = 'test';
const wrapper = shallow(<ComplexComponent title={title} />);

describe('(Component) ComplexComponent', () => {
  describe('render()', () => {
    it('should render the component', () => {
      expect(wrapper.find('div').length).toBe(1);
    });

    it('should render the title', () => {
      expect(wrapper.find('h3').text()).toEqual(title.toUpperCase());
    });
  });
});