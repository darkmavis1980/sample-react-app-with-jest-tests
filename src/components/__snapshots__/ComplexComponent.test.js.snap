// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ComplexComponent render() should render the component 1`] = `
<div>
  <h3>
    TEST
  </h3>
  <button
    className="ui button"
    id="click-me-btn"
    onClick={[Function]}
  >
    Click me!
  </button>
  <button
    className="ui button"
    id="decrease-btn"
    onClick={[Function]}
  >
    -
  </button>
  <button
    className="ui button"
    id="increase-btn"
    onClick={[Function]}
  >
    +
  </button>
  <p>
    0
  </p>
</div>
`;
